Apps/Src/ReadTritium.c

```diff
@@ -29,21 +31,39 @@ tritium_error_code_t MotorController_getTritiumError(void){
 	return T_NONE;
 }
 
+
+static void callback_hallError(void){
+        static uint8_t hall_fault_cnt = 0; //trip counter 
+        if(hall_fault_cnt > RESTART_THRESHOLD){ //try to restart the motor a few times and then fail out 
+            exception_t hallErrorPrio1 = {1, "hall sensor error prio 2", NULL};
```

Comment:
We were changing exceptions based on whether we had exceeded the restart threshold and needed to enter a nonrecoverable fault or not, but we can move this into the assert function so that we can just assert one exception or the other immediately.